codelab/shunbao_codelab/MagicNumberConverterTest.java

```diff
@@ -0,0 +1,16 @@
+package shunbao_codelab;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class MagicNumberConverterTest {
+    @Test
+    public void testConvertBinaryNumber() {
+        MagicNumberConverter convert = new MagicNumberConverter();
+        int result = convert.binaryToDecimal("0");
+        assertEquals(0, result);
+
+        assertEquals(3, convert.binaryToDecimal("11"));
```

Comment:
please also test invalid cases such as empty string, and also test negative number like "-11". Please test long number such as "11001111". 

Testing super long number to see if there is overflow: "11111111111111"